sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -323,7 +323,15 @@ ssize_t coap_handle_req(coap_pkt_t *pkt, uint8_t *resp_buf, unsigned resp_buf_le
             continue;
         }
 
-        int res = strcmp((char *)uri, resource->path);
+        int res;
```

Comment:
Seeing that these lines are now duplicate, how about factoring it into a shared function?

Reply:
Maybe at that point a custom strcmp would make sense. It could do strncmp, but if it arrives at the end of resource->path, if up to that point the strings matched and the current path position is '/', it considers that as a match.
Unless there is a use cases for prefixes that are not a whole subtree, that would remove the need for the method flag altogether.

Reply:
> do you mean that the comparison in nanocoap is the same as in gcoap?

yes!

> I like an explicit flag to qualify the search. Maybe we'll need other ways in the future to qualify a path, like with a simple regex.

ok, fine with that.

Reply:
Yes, agree on addition of a common search function for gcoap and nanocoap. As you described in another comment, this function also should be reuseable for a subtree search.

Reply:
I've compared the resource lookup functions in gcoap and nanocoap. On second thought, I don't think it's worth expanding the API to create a public function just to compare the request path to the resource path. If we do such a thing, I would be in favor of a second header file, something like `nanocoap_private.h`, or at least including a clear warning that this function isn't really useful for an application user. I'm also OK with leaving the code as is.

On the other hand, I do think it's useful to expand the API for a function that loops through an entire array of resources to look for a match, but I think that function is out of scope for this PR.

Reply:
How about ```coap_match_path(resource, url)```? Just to remove the duplicated lines for this.

Reply:
As you know, I have been working to reduce duplication between nanocoap and gcoap. At the same time, the size of the API has grown with the addition of options. Build the source doc for #11057 and review the nanocoap page to see what I mean.

I stand by my previous comment. I don't see the value of this function to a user, but I'm OK with a private API to remove the duplication.

Reply:
@kaspar030 @kb2ma I moved the matching function to a `nanocoap_private.h` header as suggested.

Reply:
I'd like to be sure there is no misunderstanding. Why did you inline the implementation in the header file?

Reply:
Yeah, I guess it does not really make any sense to have it defined in the header file. I will move it to `nanocoap.c`